activejob/lib/active_job/arguments.rb

```diff
@@ -31,6 +31,7 @@ module Arguments
     # serialized element by element. All other types are serialized using
     # GlobalID.
     def serialize(arguments)
+      raise SerializationError.new("Arguments must be an array") unless arguments.is_a?(Array)
```

Comment:
I don't think we need this here. It is just adding runtime overhead to every single argument.

Reply:
OK, removed